modules/learn/pages/data/transactions.adoc

```diff
@@ -0,0 +1,56 @@
+= Distributed ACID Transactions
+
+[abstract]
+_Distributed ACID Transactions_ guarantee that multiple documents can be updated atomically.
+
+[#transactions-and-couchbase-server]
+== Transactions and Couchbase Server
+
+_Transactions_ are operations that ensure that when multiple discrete data-elements need to be modified such that only the successful modification of all justifies the modification of any, _either_ all the modifications do occur successfully; _or_ none of them occurs.
+Transactions are implemented by Couchbase Server as _Distributed ACID Transactions_.
+
+A Distributed ACID Transaction can be performed on any number of documents.
+For each document, the transaction makes a copy of the document-data, and saves the copy temporarily in the xref:learn:data/extended-attributes.adoc[Extended Attributes] area of the document's meta data.
+Updates and modifications are then made to the key-value pairs contained by this meta data copy: consequently, the changes that occur during the transaction prior to commitment &#8212; these constituting _uncommitted_ (or _dirty_) data &#8212; are non-readable by any operation other than this transaction itself.
+Only following _commit_ of the documents modified by the transaction &#8212; when the modified data-copy is removed from the meta data area, and is written over the main body of the document's data &#8212; do the corresponding changes become readable by other transactions and operations.
+This isolation-level is known as https://jepsen.io/consistency/models/read-committed[Read Committed].
+
+Note that this use of a document's extended attributes area results in an approximate doubling of its pre-transaction size, while the transaction is underway.
+In consequence, transactions can only be used on documents whose maximum size is 10 MB (which is half of the maximum-permitted document-size, 20 MB).
+
+As well as making use of individual documents' extended attributes, Distributed ACID Transactions also create additional documents in each bucket that they access.
+These include _Active Transaction Records_ (up to 1024 of which can exist per bucket, and whose names are prefixed with `atr-`), and _Transaction Client Records_ (one of which exists per bucket, duly named `txn-client-record`).
+These documents are automatically maintained by Couchbase Server, and should not be modified by any application.
+They have no impact on use of either the _Query Service_ or the _Search Service_.
+Note that transaction records are not written to nodes that do not contain data to be updated: consequently, only nodes that _do_ contain data to be updated are affected by a transaction.
+
+No two Distributed ACID Transactions can access the same document at the same time.
+If such an attempt occurs, only one of the transactions can proceed; while the other must await commitment of the data, before itself being able to modify it.
```

Comment:
Well 'access' as in reads is fine, it's two concurrent writes that fails.
The language here makes it sound like this is a limitation or problem I think?  Detecting write-write conflicts is a good thing :) We could instead here discuss how such conflicts are handled transparently and automatically for the app, through retrying the transaction (this is a benefit of ours over other some competitor implementations).

Reply:
How about: 'Multiple Distributed ACID Transactions can read the same document at the same time. If two transactions simultaneously attempt to write to a single document, the conflict is handled transparently; by allowing one to proceed, and retrying the other.' Something like that?